tests/testthat/test-compare-character.R

```diff
@@ -71,3 +71,11 @@ test_that("not all lines are shown", {
   expect_equal(lines[1], "1/1 mismatches")
   expect_equal(length(lines), 8)
 })
+
+test_that("longer vectors (#513)", {
+  expect_warning(comp <- compare(letters[1:6], LETTERS[1:6]), NA)
+  expect_equal(
+    comp,
+    structure(list(equal = FALSE, message = "6/6 mismatches\nx[1]: \"a\"\ny[1]: \"A\"\n\nx[2]: \"b\"\ny[2]: \"B\"\n\nx[3]: \"c\"\ny[3]: \"C\"\n\nx[4]: \"d\"\ny[4]: \"D\"\n\nx[5]: \"e\"\ny[5]: \"E\""), class = "comparison")
```

Comment:
I ended up just testing each component equally, only because I do not know where to put the reference file for `expect_output_file`.  I didn't see any uses of `expect_output_file` in the tests that weren't just tempfiles.  If you don't like what I just committed just tell me where you want me to save the file for use with `expect_output_file`.